src/App.js

```diff
@@ -1,16 +1,74 @@
 import React from 'react';
+import TodoList from './components/TodoComponents/TodoList';
+import TodoForm from './components/TodoComponents/TodoForm';
+import './components/TodoComponents/Todo.css'
 
 class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
+  constructor() {
+    super();
+    this.state = {
+      todos: [],
+      task: '',
+    };
+  }
+
+  formHandler = e => {
+    this.setState({
+      [e.target.name]: e.target.value
+    });
+  }
+
+  addTodo = e => {
+    e.preventDefault();
+    if (this.state.task === '') return;
+    const newTodo = {
+      task: this.state.task,
+      id: Date.now(),
+      completed: false,
+    }
+    this.setState({
+      todos: [...this.state.todos, newTodo],
+      task: '',
+    });
+  }
+
+  listClick = e => {
+    let clickId = parseFloat(e.target.id);
+    this.state.todos.forEach(item => {
+      if (clickId === item.id) {
+        item.completed = !item.completed;
+      } 
+    });
+    e.target.classList.toggle('completed');
+  }
+
+  clearTodo = e => {
+    e.preventDefault();
+    console.log('cleartodo');
+    this.setState({
+      todos: this.state.todos.filter(item => item.completed === false)
+    });
+  }
+
   render() {
     return (
-      <div>
-        <h2>Welcome to your Todo App!</h2>
+      <div className="list-container">
+        <h2>DOING IT</h2>
+
+        <div className="todo-list" onClick={this.listClick}> 
+          <TodoList
+            todos={this.state.todos}
+          />
+        </div>
+
+        <TodoForm
+          onSubmit={this.addTodo}
+          onChange={this.formHandler}
+          value={this.state.task}
+          clearTodo={this.clearTodo}
```

Comment:
properly passed down to the child component as props, but I'd consider having the same name for each method between it's expression here and it's invocation in the child, just to avoid potential confusion. if you have end up writing close to a dozen methods and have to pass them each down, you'll want them to have the same name no matter which component they're in